doc/readme.md

```diff
@@ -0,0 +1,335 @@
+- Game-Loop: Die Game-Loop ist die wichtigste Komponente des Spieles. Sie ist eine Endlosschleife, welche einmal pro Frame aufgerufen wird. Das Spiel läuft in 30 FPS (also 30 frames per seconds), die Game-Loop wird also 30-mal in der Sekunde aufgerufen. Alle Aktionen, die wiederholt ausgeführt werden müssen, wie zum Beispiel das Bewegen und Zeichnen von Figuren, müssen innerhalb der Game-Loop stattfinden. Das Framework ermöglicht es Ihnen, eigene Aktionen in die Game-Loop zu integrieren. Wie genau das geht, erfahren Sie im Laufe dieser Anleitung.
+*Hinweis: Die Game-Loop wird automatisch ausgeführt, Sie müssen sie nicht aktiv aufrufen.*
+
+Zusätzlich existieren noch eine Vielzahl an weiteren Hilfsklassen, mit denen Sie mal mehr oder mal weniger Kontakt haben werden.
+
+### Game
+
+Game implementiert einige wichtige Methoden:
+
+- `setup` wird zu Beginn der Anwendung aufgerufen. In dieser Methode werden die Objekte (wie die Systeme) initialisiert und konfiguriert, welche bereits vor dem Spielstart existieren müssen. In der Vorgabe wird hier bereits das erste Level geladen, die Systeme angelegt und der Herd initialisert.
+- `render` Ruft die Logiken der Systeme auf.
+- `onLevelLoad` wird immer dann aufgerufen, wenn ein Level geladen wird. Hier werden später Entitäten erstellt, die initial im Level verteilt werden.
+- `frame` wird in jedem Frame einmal aufgerufen.
+- `main` startet das Spiel.
+
+- `entities`
+-entitiestoadd
+-entitiestoremove
+
+### Component
+
+- Components speichern immer die Entität zu der sie gehören
+- alle bereits implementierten Components finden Sie [hier](/ecs/components/readme.md)
+- um eigene Components zu schreiben, leiten Sie von der eigentlichen Component Klasse ab
+- siehe [create_own_content.md](ecs/create_own_content.md)
+
+### Entity
+
+- Entitäten sind leere Container und speichern Components
+- um ein Component einer Entität zu erhalten, benutzen Sie die Methode `entity#getComponent(component.class klass)`
+    - Anmerkung: Sie erhalten ein `Optional zurück`
+- siehe [create_own_content.md](doc/ecs/create_own_content.md)
+
+### System
+
+- um eigene Components zu schreiben, leiten Sie von der ECS_Systems Klasse ab
+- in der `ECS_System#update()` iterieren die Systeme über alle Entitäten mit bestimmten Components und agieren darauf
+- alle bereits implementierten Systeme finden Sie [hier](ecs/systems/readme.md)
+- siehe [create_own_content.md](ecs/create_own_content.md)
+
+
+## Übung: Eigenen Helden erstellen
+
+Zwar gibt es in den Vorlagen bereits einen Helden (den schauen wir uns am Ende dieses Kapitels genauer an), trotzdem wird Ihnen hier erklärt, wie Sie Ihre erste eigene Entität in das Spiel implementieren.
+
+*Hinweis: Wenn Sie mitprogrammieren wollen, löschen Sie die Klasse `ecs.entities.Hero` und löschen Sie in `starter.Game` die Zeile 118 (`hero = new Hero(new Point(0, 0));`) und das Import-Statement `import ecs.entities.Hero;`.*
```

Comment:
Eigentlich reicht das Initialisieren des Helden in Zeile 118, kein Grund die existierende Helden-Klasse gleich zu löschen?! Der neue Held bekommt ja sowieso einen anderen Klassennamen ...

Reply:
Und bitte nicht den kompletten Hinweis in Italics, nur das Wort "Hinweis" selbst.